ops/framework.py

```diff
@@ -1179,6 +1219,8 @@ def __le__(self, other):
             return self._under <= other._under
         elif isinstance(other, collections.abc.Set):
             return self._under <= other
+        elif isinstance(other, set):
+            return self._under <= other
```

Comment:
This seems redundant; isinstance(set(), collections.abc.Sequence) returns true, thus the existing clause should catch actual sets as well.  I'd drop the separate if for explicit sets here, as well as the other added occurrences.